.github/workflows/deploy-pbf-data-loader.yml

```diff
@@ -1,36 +1,266 @@
 
 name: deploy-pbf-data-loader
 
+env:
+  OPENSHIFT_SERVER: ${{ secrets.C5839F_OPENSHIFT_SERVER }}
+  OPENSHIFT_TOKEN: ${{ secrets.C5839F_OPENSHIFT_GITHUB_ACTIONS_SA_TOKEN }}
+  IMAGE_NAME: pbf-data-loader
+  DEFAULT_NAMESPACE: "c5839f-tools"
+  VERSION_PREFIX: "release-1.0-"
+  OPENSHIFT_NAMESPACE:
+  IMAGE_ID:
+
 # Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the "main" branch
-  push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
-
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
+    inputs:
+      Environment:
+        type: choice
+        description: Select an environment to deploy
+        options:
+          - dev
+          - uat
+          - production
+      Branch:
+        description: "Enter branch name"
+        required: true
+        default: 7127-create-env
+      Release:
+        description: "Enter build number"
+        required: true
+        default: 1
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  compute:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    outputs:
+      VERSION: ${{env.VERSION}}
+      OPENSHIFT_NAMESPACE: ${{env.OPENSHIFT_NAMESPACE}}
+      DEPLOY_SUFFIX: ${{env.DEPLOY_SUFFIX}}
+      DEVOPS_DIR: ${{env.DEVOPS_DIR}}
+      IMAGE_ID: ${{env.IMAGE_ID}}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # 1. Login to OpenShift
+      - name: Log in to OpenShift
+        uses: redhat-actions/oc-login@v1
+        with:
+          openshift_server_url: ${{ env.OPENSHIFT_SERVER }}
+          openshift_token: ${{ env.OPENSHIFT_TOKEN }}
+          insecure_skip_tls_verify: true
+          namespace: ${{ env.DEFAULT_NAMESPACE }}
+
+      # 2. Find the namespace and image digest from input parameters
+      - name: Compute namespace
+        id: set-namespace
+        uses: actions/github-script@v5
+        with:
+          script: |
+            if (`${{ github.event.inputs.Environment }}` == 'dev') {
+              return "c5839f-dev"
+            }
+            if (`${{ github.event.inputs.Environment }}` == 'uat') {
+              return "c5839f-test"
+            }
+            if (`${{ github.event.inputs.Environment }}` == 'production') {
+              return "c5839f-prod"
+            }
+
+      - name: Compute deployment env suffix
+        id: set-deploy-suffix
+        uses: actions/github-script@v5
+        with:
+          script: |
+            if (`${{ github.event.inputs.Environment }}` == 'dev') {
+              return "-dev"
+            }
+            if ( `${{ github.event.inputs.Environment }}` == 'uat') {
+              return "-uat"
+            }
+            if (`${{ github.event.inputs.Environment }}` == 'production') {
+              return ""
```

Comment:
Is this supposed to be blank for production?

Reply:
there is no suffix for Prod resources, so this is left blank.